kernel/driver.cc

```diff
@@ -250,292 +204,231 @@ int main(int argc, char **argv)
-		printf("\n");
-		printf("    -m module_file\n");
-		printf("        load the specified module (aka plugin)\n");
-		printf("\n");
-		printf("    -X\n");
-		printf("        enable tracing of core data structure changes. for debugging\n");
-		printf("\n");
-		printf("    -M\n");
-		printf("        will slightly randomize allocated pointer addresses. for debugging\n");
-		printf("\n");
-		printf("    -A\n");
-		printf("        will call abort() at the end of the script. for debugging\n");
-		printf("\n");
-		printf("    -r <module_name>\n");
-		printf("        elaborate command line arguments using the specified top module\n");
-		printf("\n");
-		printf("    -D <macro>[=<value>]\n");
-		printf("        set the specified Verilog define (via \"read -define\")\n");
-		printf("\n");
-		printf("    -P <header_id>[:<filename>]\n");
-		printf("        dump the design when printing the specified log header to a file.\n");
-		printf("        yosys_dump_<header_id>.il is used as filename if none is specified.\n");
-		printf("        Use 'ALL' as <header_id> to dump at every header.\n");
-		printf("\n");
-		printf("    -W regex\n");
-		printf("        print a warning for all log messages matching the regex.\n");
-		printf("\n");
-		printf("    -w regex\n");
-		printf("        if a warning message matches the regex, it is printed as regular\n");
-		printf("        message instead.\n");
-		printf("\n");
-		printf("    -e regex\n");
-		printf("        if a warning message matches the regex, it is printed as error\n");
-		printf("        message instead and the tool terminates with a nonzero return code.\n");
-		printf("\n");
-		printf("    -E <depsfile>\n");
-		printf("        write a Makefile dependencies file with in- and output file names\n");
-		printf("\n");
-		printf("    -x <feature>\n");
-		printf("        do not print warnings for the specified experimental feature\n");
-		printf("\n");
-		printf("    -g\n");
-		printf("        globally enable debug log messages\n");
-		printf("\n");
-		printf("    -V\n");
-		printf("        print version information and exit\n");
-		printf("\n");
-		printf("The option -S is a shortcut for calling the \"synth\" command, a default\n");
-		printf("script for transforming the Verilog input to a gate-level netlist. For example:\n");
-		printf("\n");
-		printf("    yosys -o output.blif -S input.v\n");
-		printf("\n");
-		printf("For more complex synthesis jobs it is recommended to use the read_* and write_*\n");
-		printf("commands in a script file instead of specifying input and output files on the\n");
-		printf("command line.\n");
-		printf("\n");
-		printf("When no commands, script files or input files are specified on the command\n");
-		printf("line, yosys automatically enters the interactive command mode. Use the 'help'\n");
-		printf("command to get information on the individual commands.\n");
-		printf("\n");
+	cxxopts::Options options(argv[0], "Yosys Open SYnthesis Suite");
+
+	options.add_options("functional")
```

Comment:
Suggestion: "operation" for the name of the section, and include all the options from the logging section which set the purpose of calling the binary, which is `-H`, `-h`, `-V`